src/IRremoteESP8266.h

```diff
@@ -1,1051 +1,1062 @@
+#endif
+
+// Use millisecond 'delay()' calls where we can to avoid tripping the WDT.
+// Note: If you plan to send IR messages in the callbacks of the AsyncWebserver
+//       library, you need to set ALLOW_DELAY_CALLS to false.
+//       Ref: https://github.com/crankyoldgit/IRremoteESP8266/issues/430
+#ifndef ALLOW_DELAY_CALLS
+#define ALLOW_DELAY_CALLS true
+#endif  // ALLOW_DELAY_CALLS
+
+// Enable a run-time settable high-pass filter on captured data **before**
+// trying any protocol decoding.
+// i.e. Try to remove/merge any really short pulses detected in the raw data.
+// Note: Even when this option is enabled, it is _off_ by default, and requires
+//       a user who knows what they are doing to enable it.
+//       The option to disable this feature is here if your project is _really_
+//       tight on resources. i.e. Saves a small handful of bytes and cpu time.
+// WARNING: If you use this feature at runtime, you can no longer trust the
+//          **raw** data captured. It will now have been slightly **cooked**!
+// DANGER: If you set the `noise_floor` value too high, it **WILL** break
+//         decoding of some protocols. You have been warned. Here Be Dragons!
+//
+// See: `irrecv::decode()` in IRrecv.cpp for more info.
+#ifndef ENABLE_NOISE_FILTER_OPTION
+#define ENABLE_NOISE_FILTER_OPTION true
+#endif  // ENABLE_NOISE_FILTER_OPTION
+/*
+ * Always add to the end of the list and should never remove entries
+ * or change order. Projects may save the type number for later usage
+ * so numbering should always stay the same.
+ */
+enum decode_type_t {
+  UNKNOWN = -1,
+  UNUSED = 0,
+  RC5,
+  RC6,
+  NEC,
+  SONY,
+  PANASONIC,  // (5)
+  JVC,
+  SAMSUNG,
+  WHYNTER,
+  AIWA_RC_T501,
+  LG,  // (10)
+  SANYO,
+  MITSUBISHI,
+  DISH,
+  SHARP,
+  COOLIX,  // (15)
+  DAIKIN,
+  DENON,
+  KELVINATOR,
+  SHERWOOD,
+  MITSUBISHI_AC,  // (20)
+  RCMM,
+  SANYO_LC7461,
+  RC5X,
+  GREE,
+  PRONTO,  // Technically not a protocol, but an encoding. (25)
+  NEC_LIKE,
+  ARGO,
+  TROTEC,
+  NIKAI,
+  RAW,  // Technically not a protocol, but an encoding. (30)
+  GLOBALCACHE,  // Technically not a protocol, but an encoding.
+  TOSHIBA_AC,
+  FUJITSU_AC,
+  MIDEA,
+  MAGIQUEST,  // (35)
+  LASERTAG,
+  CARRIER_AC,
+  HAIER_AC,
+  MITSUBISHI2,
+  HITACHI_AC,  // (40)
+  HITACHI_AC1,
+  HITACHI_AC2,
+  GICABLE,
+  HAIER_AC_YRW02,
+  WHIRLPOOL_AC,  // (45)
+  SAMSUNG_AC,
+  LUTRON,
+  ELECTRA_AC,
+  PANASONIC_AC,
+  PIONEER,  // (50)
+  LG2,
+  MWM,
+  DAIKIN2,
+  VESTEL_AC,
+  TECO,  // (55)
+  SAMSUNG36,
+  TCL112AC,
+  LEGOPF,
+  MITSUBISHI_HEAVY_88,
+  MITSUBISHI_HEAVY_152,  // 60
+  DAIKIN216,
+  SHARP_AC,
+  GOODWEATHER,
+  INAX,
+  DAIKIN160,  // 65
+  NEOCLIMA,
+  DAIKIN176,
+  DAIKIN128,
+  AMCOR,
+  DAIKIN152,  // 70
+  MITSUBISHI136,
+  MITSUBISHI112,
+  HITACHI_AC424,
+  SONY_38K,
+  EPSON,  // 75
+  SYMPHONY,
+  HITACHI_AC3,
+  DAIKIN64,
+  AIRWELL,
+  DELONGHI_AC,  // 80
+  DOSHISHA,
+  MULTIBRACKETS,
+  CARRIER_AC40,
+  CARRIER_AC48,

```

Comment:
Typically, new protocols are always added last here

Reply:
Correct. You must not change the order/position of existing Protocols in this `enum`. It defines the number via sequence. If you change the order, you change the numbers of other protocols. Some programs/tools rely on that number being the same.

Reply:
Correct. New protocols must go at the end.

Reply:
fixed this